# Import required libraries

In [39]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import confusion_matrix

# READ the `dataset.csv` file into a dataframe and drop NaN column 

In [8]:
df = pd.read_csv("dataset.csv")
df.drop(df.columns[6], axis = 1, inplace=True) 
df.head(5)

Uninfected  14493.5    0  0.1  0.2  0.3
0  Uninfected  13663.5  0.0  0.0  0.0  0.0
1  Uninfected  14300.0  0.0  0.0  0.0  0.0
2  Uninfected   7735.5  0.0  0.0  0.0  0.0
3  Uninfected  11123.0  0.0  0.0  0.0  0.0
4  Uninfected   9863.0  0.0  0.0  0.0  0.0

# Add labels to the dataframe

In [9]:
headers = ['Cell_diagnosis','area0','area1','area2','area3','area4']
print(headers)
df.columns = headers
df.head(5)

['Cell_diagnosis', 'area0', 'area1', 'area2', 'area3', 'area4']


Cell_diagnosis    area0  area1  area2  area3  area4
0     Uninfected  13663.5    0.0    0.0    0.0    0.0
1     Uninfected  14300.0    0.0    0.0    0.0    0.0
2     Uninfected   7735.5    0.0    0.0    0.0    0.0
3     Uninfected  11123.0    0.0    0.0    0.0    0.0
4     Uninfected   9863.0    0.0    0.0    0.0    0.0

# Drop the `Cell_diagnosis` column as it is a string column and not required for the training

In [10]:
x = df.drop(["Cell_diagnosis"],axis=1)
y = df["Cell_diagnosis"]
x_train, x_test, y_train, y_test = train_test_split(x,y,test_size=0.2,random_state=42)

# Generate a RandomForestClassifier model with different chosen estimators and depth and train the model with the train data

In [27]:
model = RandomForestClassifier(n_estimators=100,max_depth=5)
model.fit(x_train,y_train);

# Make predictions by feeding the test data into the model created and observe the predictions

In [40]:
predictions = model.predict(x_test)

print('Eval:', metrics.classification_report(predictions,y_test))
print('Model score:', model.score(x_test,y_test))
print('\nConfusion matrix: \n',confusion_matrix(y_test, predictions))

Eval:               precision    recall  f1-score   support

 Parasitized       0.89      0.90      0.90      2667
  Uninfected       0.91      0.90      0.90      2845

    accuracy                           0.90      5512
   macro avg       0.90      0.90      0.90      5512
weighted avg       0.90      0.90      0.90      5512

Model score: 0.9011248185776488

Confusion matrix: 
 [[2407  285]
 [ 260 2560]]


# Model has resulted in `90% accuracy`

## The confusion matrix shows that 2407 positive (parasitized) values were predicted correctly as positivewhile 260 possitve values were incorectly predicted as negative(uninfected)

## As for the uninfected values, 2560 values were predicted correctly as negative(uninfected) while 285 values were incorrectly predicted as positive(parasitized)